src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/BuildUpToDateCheck.cs

```diff
@@ -118,12 +120,27 @@ private void OnProjectChanged(IProjectSubscriptionUpdate e)
             _isDisabled = disableFastUpToDateCheckString != null && string.Equals(disableFastUpToDateCheckString, TrueValue, StringComparison.OrdinalIgnoreCase);
 
             _msBuildProjectFullPath = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildProjectFullPathProperty, _msBuildProjectFullPath);
-            foreach (var referenceSchema in ReferenceSchemas)
+            _markerFile = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.CopyUpToDateMarkerProperty, null);
```

Comment:
That might actually make it easier i think. https://github.com/dotnet/project-system/pull/2416 is making it so that we just query the projectitemschemaservice for all item types and we can add CopyUpToDateMarker item to the schema. If we did that, do we still need the CheckMarkers method below? Ideally if we can avoid any special code here for the markers that would be ideal.